test/E2ETests/Smoke.E2E.Tests/SuiteLifeCycleTests.cs

```diff
@@ -27,348 +27,279 @@ private void ValidateTestRunLifecycle(string targetFramework)
-            Derived class TestInitialize was called
-            Derived class TestMethod was called
-            Derived class TestCleanup was called
-            TestCleanup was called
-            DisposeAsync was called
-            Dispose was called
-            """ : """
-            Ctor was called
-            Derived class Ctor was called
-            TestInitialize was called
-            Derived class TestInitialize was called
-            Derived class TestMethod was called
-            Derived class TestCleanup was called
-            TestCleanup was called
-            Dispose was called
+        caseInheritClassWithCleanupInheritanceBehaviorNone.Messages.Single().Text.Should().Contain(
+            $"""
+            Current ClassInitialize was called
+            Base Ctor was called
+            Current Ctor was called
+            Base TestInitialize was called
+            Current TestInitialize was called
+            Current TestMethod was called
+            Current TestCleanup was called
+            Base TestCleanup was called
+            {(targetFramework == "net6.0" ? "Base DisposeAsync was called\r\nBase Dispose was called" : "Base Dispose was called")}
             """);
 
         // Test the parent test method.
         var caseInheritClassWithCleanupInheritanceBehaviorNone_ParentTestMethod = RunEventsHandler.PassedTests.Single(x => x.TestCase.FullyQualifiedName.Contains("SuiteLifeCycleTestClass_InheritClassWithCleanupInheritanceBehaviorNone.TestMethod"));
+        numberOfLines = caseInheritClassWithCleanupInheritanceBehaviorNone_ParentTestMethod.Messages.Single().Text.Split('\n').Length;
+        Verify(numberOfLines == (targetFramework == "net6.0" ? 25 : 24)); // The number of the logs + 3 empty lines + 1 for the logs' header.
         caseInheritClassWithCleanupInheritanceBehaviorNone_ParentTestMethod.Outcome.Should().Be(Microsoft.VisualStudio.TestPlatform.ObjectModel.TestOutcome.Passed);
-        caseInheritClassWithCleanupInheritanceBehaviorNone_ParentTestMethod.Messages.Single().Text.Should().Contain(targetFramework == "net6.0" ? """
-            Ctor was called
-            Derived class Ctor was called
-            TestInitialize was called
-            Derived class TestInitialize was called
-            TestMethod was called
-            Derived class TestCleanup was called
-            TestCleanup was called
-            DisposeAsync was called
-            Dispose was called
-            """ : """
-            Ctor was called
-            Derived class Ctor was called
-            TestInitialize was called
-            Derived class TestInitialize was called
-            TestMethod was called
-            Derived class TestCleanup was called
-            TestCleanup was called
-            Dispose was called
-            """);
+        caseInheritClassWithCleanupInheritanceBehaviorNone_ParentTestMethod.Messages.Single().Text.Should().Contain(
+            targetFramework == "net6.0"
+            ? """
```

Comment:
So this test case is failing for net6.0. As far as I can see, messages are not produced in the same order.

Potential causes could be:
- Parallel execution working differently for netcore and netfx causing different methods to be called at different time
- Methods are not discovered in the same order
- Implementation is buggy